NOTES.md

```diff
@@ -0,0 +1,54 @@
+There are two directories here, `nokogiri` and `homebaked` that differ in their XML builders... and on other axes.
+
+# Shared features
+
+- API errors transformed into exceptions.
+- Sessioning handled with a block.
+- For demo, credentials from a git-ignored config.yaml.
+- `verbose` flag for debugging and easy demos.
+
+# Differences
+
+`nokogiri`:
+```
+  fragment = Nokogiri::XML::DocumentFragment.parse("")
+  Nokogiri::XML::Builder.with(fragment) do |xml|
+    xml.id id
+    xml.meta {
+      xml.send("mf-note") {
+        xml.note "Hello"
+      }
+    }
+  end
+  args_xml = fragment.to_xml
+  mf.call("asset.set", args_xml)
+```
+
+`homebaked`:
+```
+  mf.set_asset id: id, meta: {mf_note: {note: "Hello"}}
```

Comment:
While the main application will probably be Rails, I think there's a goal to make tools that are reusable. The client should be small, and it shouldn't need to depend on an entire web framework. I think [Builder](https://github.com/jimweirich/builder) may have originated in Rails, so it might give us the same functionality? I'm not sure.

It's unclear from me from the [docs](https://api.rubyonrails.org/classes/Array.html#method-i-to_xml) how `to_xml` supports the generation of elements.

In your example we would still need to extract the xml fragment from inside `<hash>`... That's doable, but it's more complexity.